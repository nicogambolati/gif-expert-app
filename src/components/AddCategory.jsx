import { useState } from 'react';

// export const AddCategory = ({ setCategories }) => {
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("One Punch");

  /**
   ** Siempre estoy recibiendo el event,despues lo Desestructuramos
   **/
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    /**
     ** Chequea que haya mas de un  caracter
     **/
    if (inputValue.trim().length <= 1) return;
    console.log(inputValue);

    /**
     ** Todas las categorias en el estado actual,
     ** insertamos el nuevo inputValue seguido de la desestructuracion de las categorias anteriores
     **/
    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="test"
        placeholder="Buscar gifs"
        onChange={onInputChange}
        // value={inputValue}
        //   onChange={(event) => onInputChange(event)}
      />
    </form>
  );
};
