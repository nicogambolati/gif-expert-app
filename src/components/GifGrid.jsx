import { GifItem } from './GifItem';
import { useFetchGifts } from '../hooks/useFetchGifts';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifts(category);
  console.log({ isLoading });

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && (<h2>Cargando...</h2>)
      }


      <div className="card-grid">
        {images.map((image) => {
          return (
            <GifItem
              key={image.id}
              /*Esparcimos todas las props que vienen dentro de image*/
              {...image}
            />
          );
        })}
      </div>
    </>
  );
};
