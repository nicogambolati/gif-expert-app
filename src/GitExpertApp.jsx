import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GitExpertApp = () => {
  const [categories, setCategories] = useState(["Dragon Ball Z"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GitExpertApp</h1>

      <AddCategory onNewCategory={(event) => onAddCategory(event)} />

      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
};
