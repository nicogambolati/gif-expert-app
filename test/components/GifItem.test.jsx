import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("test in <GifItem />", () => {
  const title = "Meme";
  const url = "https://gif-search.com/algo.jpg";

  test("should match snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("It must show the image with the URL and ALT indicated", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();

    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("must show the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
