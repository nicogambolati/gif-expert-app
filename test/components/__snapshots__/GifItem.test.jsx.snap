// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Meme"
      src="https://gif-search.com/algo.jpg"
    />
    <p>
      Meme
    </p>
  </div>
</div>
`;
